tests/bindings/python/test_reset.py

```diff
@@ -0,0 +1,144 @@
+#
+# Tests the bindings for OrderedEntity, which is the base class for all
+# entities that can be ordered relative to one another.
+#
+import unittest
+
+
+class OrderedEntityTestCase(unittest.TestCase):
+
+    def test_create_destroy(self):
+        from libcellml import Reset
+
+        x = Reset()
+        y = Reset()
+        z = Reset(y)
+        del(x, y, z)
+
+    def test_set_get_variable(self):
+        from libcellml import Reset
+        from libcellml import Variable
+
+        r = Reset()
+        v = Variable()
+        v.setName("glucose")
+
+        self.assertEqual(None, r.getVariable())
+
+        r.setVariable(v)
+
+        self.assertEqual("glucose", r.getVariable().getName())
+
+    def test_add_when(self):
+        from libcellml import Reset
+        from libcellml import When
+
+        # addWhen()
+        w = When()
+        x = Reset()
+        x.addWhen(w)
+        self.assertEqual(1, x.whenCount())
+
+    def test_remove_when(self):
+        from libcellml import Reset
+        from libcellml import When
+
+        w = When()
+        r = Reset()
+
+        r.addWhen(w)
+
+        self.assertEqual(1, r.whenCount())
+
+        self.assertTrue(r.removeWhen(0))
+        self.assertEqual(0, r.whenCount())
+
+        r.addWhen(w)
+
+        self.assertTrue(r.removeWhen(w))
+        self.assertEqual(0, r.whenCount())
```

Comment:
What is the rationale for the 3 lines of code above? It looks like a repeat of what is above?

Reply:
Different method used to remove the when one by index the other by class.

Reply:
Ok.